paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -576,6 +576,25 @@ public void renameTag(String tagName, String targetTagName) {
         tagManager().renameTag(tagName, targetTagName);
     }
 
+    @Override
+    public void replaceTag(String tagName, @Nullable Duration timeRetained) {
+        Snapshot latestSnapshot = snapshotManager().latestSnapshot();
+        SnapshotNotExistException.checkNotNull(
+                latestSnapshot, "Cannot create tag because latest snapshot doesn't exist.");
+        tagManager()
+                .replaceTag(latestSnapshot, tagName, timeRetained, store().createTagCallbacks());
+    }
+
+    @Override
+    public void replaceTag(String tagName, long fromSnapshotId, @Nullable Duration timeRetained) {
+        tagManager()
+                .replaceTag(
+                        findSnapshot(fromSnapshotId),
+                        tagName,
+                        timeRetained,
+                        store().createTagCallbacks());
```

Comment:
Here you should think this case:
If user update a tag, but he don't want call createTagCallbacks.

Reply:
good, replace the existing tag is no need to call callback again.